languages/python/reference/exercise-concepts/matrix.md

```diff
@@ -0,0 +1,39 @@
+# Concepts of matrix  
+
+
+[Example Implementation](https://github.com/exercism/python/blob/master/exercises/matrix/example.py)
+
+[Alternate Implementation](https://exercism.io/tracks/python/exercises/matrix/solutions/e5004e990ddc4582a50ecc1f660c31df)
+
+[Extended Implmentation](https://exercism.io/tracks/python/exercises/matrix/solutions/b6a3486a35c14372b64fdc35e7c6f98f)
+
+
+ ## General  
+
+  - classes:  the exercise objective is to define a `matrix` type.  Tested methods are linked to a `matrix` class
+  - objects:  creating different instances with different data representing different `matrices` is tested
+  - constructor:  customizing object initalization with actions and persisting data.  The example uses a constructor to process the passed in data into a list of lists assigned to an instance property
+  - magic methods: the example uses the `__init__` magic method as its constructor for the class 
```

Comment:
@yawpitch does `__init__` count as a magic method? I wouldn't think so, but am open to correction.

Reply:
It's certainly the least "magic" of the dunder methods, but it's one of them, as is `__new__`.

Reply:
Hmm... should we consider it to be *not* one of them for the purpose of concept listing? Otherwise, *every* class-based exercise will also be a `magic methods` exercise. That seems wrong to me.

Reply:
As we extract these into their own files I think it'd be fine to consider `__init__` a special case.

Reply:
I struggled a bit here with both `__init__` and `self`.  While they're arguably part of constructors and objects like any other OOP language, they're also unique to python in the way they're used.  Learners fight with them.  Since `matrix` was the first exercise in V2 to introduce classes, pulling them out seems relevant here.  But having every class based exercise be tagged with `magic-methods`  would be excessive.  Shall I remove them?

Reply:
Personally I'd say over-tagging exercises with concepts isn't necessarily a bad idea, at least at this point. Both `self` and `__init__` (and `__new__`) might all make more sense in a `class-constructors` concept the documentation of which redirects to `magic-methods`, just like the various other "groupings" of magic methods may need their own concept (ie `rich-comparisons`, `operator-overloading`, `string-and-repr`, and so on).